pkg/templates/templates.go

```diff
@@ -269,6 +290,20 @@ func (t *TemplateResolver) ResolveTemplate(tmplJSON []byte, context interface{})
 		"toBool":           toBool,
 	}
 
+	// Verify encryption settings (since encryption is checked and keys are set per-policy)
+	if UsesEncryption(tmplJSON, t.config.StartDelim, t.config.StopDelim) &&
```

Comment:
Instead of performing the validation every time templates are resolved, could you please separate out the validation code from `SetEncryptionConfig` into a separate helper function and call that helper function in `SetEncryptionConfig` and `NewResolver`?

Reply:
Ah, yes that'd make more sense.